pages/[id].js

```diff
@@ -39,16 +48,19 @@ export default function Detail() {
   }
 
   return (
-    <RoastDetailCard
-      onFavorite={addToFavorites}
-      name={data.name}
-      roaster={data.roaster}
-      arabica={data.arabica}
-      robusta={data.robusta}
-      level={data.level}
-      provenance={data.provenance}
-      score={data.score.map(({ rating }) => rating)}
-      session={session}
-    />
+    <>
+      <RoastDetailCard
+        onFavorite={addToFavorites}
+        name={data.name}
+        roaster={data.roaster}
+        arabica={data.arabica}
+        robusta={data.robusta}
+        level={data.level}
+        provenance={data.provenance}
+        score={data.score.map(({ rating }) => rating)}
+        session={session}
+        isFavorite={isFavorite(id)}
```

Comment:
Then, this gets shorter, too.
```suggestion
        isFavorite={isFavorite}
```